modules/gateway/rpc.go

```diff
@@ -173,6 +173,10 @@ func (g *Gateway) threadedListenPeer(p *peer) {
 		// The handler is responsible for closing the connection, though a
 		// default deadline has been set.
 		go g.threadedHandleConn(conn)
+
+		if !g.managedSleep(peerRPCDelay) {
```

Comment:
* sorry, this ratelimit is per-peer. So one peer cannot drown out the other peers.

We have another ratelimit that allows only one peer to connect per 6 seconds, and that could potentially be drowned out by an adversary. But it only affects inbound connections. One option we thought of was to require a PoW in the handshake.